mindbender/maya/lib.py

```diff
@@ -416,6 +416,56 @@ def apply_shaders(relationships):
             cmds.sets(meshes, forceElement=shader)
 
 
+def displacement_shader_dir(nodes):
+
+    valid_nodes = cmds.ls(nodes,
+                          long=True,
+                          recursive=True,
+                          showType=True,
+                          objectsOnly=True,
+                          type="transform")
+
+    meshes = {}
+    for mesh in valid_nodes:
+        shapes = cmds.listRelatives(valid_nodes[0],
+                                    shapes=True,
+                                    fullPath=True) or list()
+
+        if shapes:
+            shape = shapes[0]
+            if not cmds.nodeType(shape):
```

Comment:
This doesn't seem to actually check anything useful. The `cmds.nodeType` command will return the node type (str) of the node and when the node will always return an actual value (so be True) otherwise if the node does not exist it will raise an error. In this case it will always evaluate as True, and will never skip a node.

You might have meant to do something like this:

```python
cmds.objectType(shape, isAType="mesh")
```

Though note that `cmds.listRelatives` also allows you to filter to types, like meshes: `cmds.listRelatives(node, shapes=True, fullPath=True, type="mesh")` (In this case there's even node need to include `shapes=True` because it already filters to the mesh type anyway.